### ISSUE TITLE
NativeAOT scanner failure

### ISSUE THREAD

**Issue by kant2002** [dotnet/runtime#65612]
I hit following exception during pretty simple EF Core app.
```
    ILCompiler.ScannerFailedException: VTable of type 'System.Collections.Generic.ICollection`1<Table1>' not computed by the IL scanner. You can work around by running the compilation with scanner disabled.
       at ILCompiler.ILScanResults.ScannedVTableProvider.GetSlice(TypeDesc type)
       at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
       at ILCompiler.DependencyAnalysis.EETypeNode.ComputeNonRelocationBasedDependencies(NodeFactory factory)
       at ILCompiler.DependencyAnalysis.ConstructedEETypeNode.ComputeNonRelocationBasedDependencies(NodeFactory factory)
       at ILCompiler.DependencyAnalysis.ObjectNode.GetStaticDependencies(NodeFactory factory)
       at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.GetStaticDependenciesImpl(DependencyNodeCore`1 node)
       at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.GetStaticDependencies(DependencyNodeCore`1 node)
       at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.ProcessMarkStack()
       at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.ComputeMarkedNodes()
       at ILCompiler.RyuJitCompilation.CompileInternal(String outputFile, ObjectDumper dumper)
       at ILCompiler.Compilation.ILCompiler.ICompilation.Compile(String outputFile, ObjectDumper dumper)
       at ILCompiler.Program.Run(String[] args)
       at ILCompiler.Program.Main(String[] args)
```
Example program which trigger the issue.

```
using Microsoft.EntityFrameworkCore;

Console.WriteLine("Building options");
var builder = new DbContextOptionsBuilder<MyDbContext>();
builder.UseSqlite("DataSource=test.db");
var options = builder.Options;

Console.WriteLine("Creating context");
var context = new MyDbContext(options);
Console.WriteLine("Recreate database");
context.Database.EnsureDeleted();
context.Database.EnsureCreated();
Console.WriteLine("Creating test data!");

// Line below trigger the issue.
context.Table1.Add(new Table1() { Id = 1 });
context.SaveChanges();

public class Table1
{
    public int Id { get; set; }
}
public class MyDbContext : DbContext
{
    public MyDbContext(DbContextOptions<MyDbContext> options) : base(options) { }
    
    public DbSet<Table1> Table1 { get; set; } = default!;
}
```
RD.xml files
- https://github.com/kant2002/RdXmlLibrary/blob/b51448e6eb67fbcf3f81e729972e83adf909d5c1/Microsoft.EntityFrameworkCore.rd.xml
- https://github.com/kant2002/RdXmlLibrary/blob/b51448e6eb67fbcf3f81e729972e83adf909d5c1/Microsoft.EntityFrameworkCore.Sqlite.rd.xml

**Comment by hez2010** [comment]
I hit this too, in an efcore app.

**Comment by MichalStrehovsky** [comment]
> pretty simple EF Core app

"Pretty simple" and "EF Core app" cannot be used in a single sentence addressed to runtime devs :).

The fix will be in #67421. A workaround is to put `Table1[]` into RD.XML.
